bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -99,28 +101,15 @@
   private final Map<MethodDescriptor<?, ?>, Predicate<?>> methodsToRetryMap =
       createMethodRetryMap();
 
-  /**
-   * Creates a SslContext.
-   */
-  public interface SslContextFactory {
-    SslContext create();
-  }
+  private final static SslContextBuilder sslBuilder = GrpcSslContexts.forClient();
 
-  public static final SslContextFactory SSL_CONTEXT_FACTORY = new SslContextFactory() {
-    @SuppressWarnings("deprecation")
-    @Override
-    public SslContext create() {
-      try {
-        // We create multiple channels via refreshing and pooling channel implementation.
-        // Each one needs its own SslContext.
-        return SslContext.newClientContext();
-      } catch (SSLException e) {
-        throw new IllegalStateException("Could not create an ssl context.", e);
-      }
+  static {
+
+    if (System.getProperty("java.version").startsWith("1.7.")) {
+      // The grpc cyphers only work in JDK 1.8+.  Use the default system cyphers for JDK 1.7.
+      sslBuilder.ciphers(null);
```

Comment:
You may also want to log a warning that they should use Java 8 to get more secure ciphers.